test/test_tubby.rb

```diff
@@ -98,6 +98,14 @@ def test_attrs
     assert_equal '<h1 c="" d="123" e="" f="1 false 2" g></h1>', tmpl.to_s
   end
 
+  def test_data_attrs
+    tmpl = Tubby.new { |t|
+      t.div(class: %w[a b], data: {attr1: 'a', attr2: 1})
+    }
+
+    assert_equal '<div class="a b" data-attr1="a" data-attr2="1"></div>', tmpl.to_s
+  end
+
```

Comment:
Can you add tests for these cases?

- `t.div(data: {attr1: 'a'}, :'data-attr2': 'b')`
- `t.div(:'data-attr2': 'b', data: {attr1: 'a'})`
- `t.div(data: {attr1: 'a'}, :'data-attr1': 'b')`
- `t.div(:'data-attr1': 'b', data: {attr1: 'a'})`

I think the most sensible behavior is "last data attribute wins", but I'm fine with any behavior as long as there's tests and it's somewhat consistent.

Reply:
Yeah, that makes perfect sense. Added test (and fixed the code so the last attribute win, and the whole attribute order would be preserved).